.github/workflows/pr_ci.yml

```diff
@@ -15,17 +15,15 @@ jobs:
     name: Build & Test
     steps:
     - uses: actions/checkout@v2
-    - uses: actions/setup-node@v2
+    - uses: actions/cache@v3
       with:
-        node-version: "15"
-    - name: Setup .NET 6.0
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.x
-        include-prerelease: true
+         path: ~/.nuget/packages
+         key: ${{ runner.os }}-nuget-${{ hashFiles('**/packages.lock.json') }}
+         restore-keys: |
+            ${{ runner.os }}-nuget-
```

Comment:
Could you do some repeated measurements before/after so we can see an average for the total workflow runtime (or cache+build)? Two GHA builds can vary wildly between two runs, so any two builds are not really comparable.

We've experimented around for NuGet and NPM caching with our [OSOCE solution](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions), which uses `OrchardCore.Application.Cms.Targets` (so basically every Orchard package) and some, so it should be comparable in terms of NuGet restore requirements. It turned out that restoring the cache under Windows is so slow that it cancels out the savings.

Reply:
I'll check with some verbose output...

Reply:
The comment I made at the end shows the results, with no restore it's faster to have the cache in use. Restore is I/O intensive even when nothing needs to be restored (NuGet cache contains items).

Reply:
If you mean [this comment](https://github.com/OrchardCMS/OrchardCore/pull/12589#issuecomment-1279934310) then I only see a single measurement. Sorry for being a pain here, it's just that a +-10% jitter in the performance of GitHub Actions builds is usual, with particularly (un)lucky ones even hitting outside of that, so while this is promising, I'd like to see with multiple runs whether the speedup persists.